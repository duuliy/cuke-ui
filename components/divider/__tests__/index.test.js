import React from "react";
import assert from "power-assert";
import { render, shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Divider from "../index";

describe("<Divider/>", () => {
  it("should render a <Divider/> components", () => {
    const wrapper = render(
      <div>
        <Divider />
        <Divider dashed />
        <Divider type="vertical" />
        <Divider>cuke ui</Divider>
        <Divider type="horizontal" position="left">
          11
        </Divider>
        <Divider position="right">cuke ui</Divider>
        <Divider position="left">cuke ui</Divider>
      </div>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("should find cuke-divider classnames", () => {
    const wrapper = render(
      <div>
        <Divider />
        <Divider dashed />
        <Divider type="vertical" />
        <Divider position="left">cuke ui</Divider>
      </div>
    );
    assert(wrapper.find(".cuke-divider").length >= 1);
    assert(wrapper.find(".cuke-divider-vertical").length === 1);
    assert(wrapper.find(".cuke-divider-dashed").length === 1);
    assert(wrapper.find(".cuke-divider-title-text").length === 1);
  });

  it("should render dashed divider", () => {
    const wrapper = shallow(<Divider dashed />);
    assert(wrapper.find(".cuke-divider-dashed").length === 1);
  });

  it("should render vertical divider", () => {
    const wrapper = shallow(<Divider type="vertical" />);
    assert(wrapper.find(".cuke-divider-vertical").length === 1);
  });

  it("should render center title divider", () => {
    const wrapper = shallow(<Divider>cuke ui title</Divider>);
    assert(wrapper.find(".cuke-divider-title-text").length === 1);
    assert(wrapper.find("b").length === 2);
    assert(wrapper.find("span").length === 1);
  });

  it("should render left title divider", () => {
    const wrapper = shallow(<Divider position="left">cuke ui title</Divider>);
    assert(wrapper.find(".cuke-divider-title-text").length === 1);
    assert(wrapper.find("b").length === 2);
    assert(wrapper.find("span").length === 1);
  });

  it("should render right title divider", () => {
    const wrapper = shallow(<Divider position="right">cuke ui title</Divider>);
    assert(wrapper.find(".cuke-divider-title-text").length === 1);
    assert(wrapper.find("b").length === 2);
    assert(wrapper.find("span").length === 1);
  });
});
