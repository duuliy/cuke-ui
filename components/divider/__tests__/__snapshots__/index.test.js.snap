// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Divider/> should render a <Divider/> components 1`] = `
<div>
  <div
    class="cuke-divider cuke-divider-horizontal cuke-divider-horizontal-title-center"
  />
  <div
    class="cuke-divider cuke-divider-horizontal cuke-divider-horizontal-title-center cuke-divider-dashed"
  />
  <div
    class="cuke-divider cuke-divider-vertical cuke-divider-horizontal-title-center"
  />
  <div
    class="cuke-divider cuke-divider-horizontal cuke-divider-horizontal-title-center cuke-divider-horizontal-title"
  >
    <b />
    <span
      class="cuke-divider-title-text"
    >
      cuke ui
    </span>
    <b />
  </div>
  <div
    class="cuke-divider cuke-divider-horizontal cuke-divider-horizontal-title-left cuke-divider-horizontal-title"
  >
    <b />
    <span
      class="cuke-divider-title-text"
    >
      11
    </span>
    <b />
  </div>
  <div
    class="cuke-divider cuke-divider-horizontal cuke-divider-horizontal-title-right cuke-divider-horizontal-title"
  >
    <b />
    <span
      class="cuke-divider-title-text"
    >
      cuke ui
    </span>
    <b />
  </div>
  <div
    class="cuke-divider cuke-divider-horizontal cuke-divider-horizontal-title-left cuke-divider-horizontal-title"
  >
    <b />
    <span
      class="cuke-divider-title-text"
    >
      cuke ui
    </span>
    <b />
  </div>
</div>
`;
